src/JustEat.StatsD/TagsFormatters/StatsDTagsFormatter.cs

```diff
@@ -11,10 +12,20 @@ namespace JustEat.StatsD.TagsFormatters
     /// </summary>
     public abstract class StatsDTagsFormatter : IStatsDTagsFormatter
     {
-        private readonly string _prefix;
-        private readonly string _suffix;
-        private readonly string _tagsSeparator;
-        private readonly string _keyValueSeparator;
+        private const int SafeUdpPacketSize = 512;
+
+        [ThreadStatic]
+        private static char[]? _buffer;
+        private static char[] Buffer => _buffer ??= new char[SafeUdpPacketSize];
+
+        private readonly char[] _prefix;
```

Comment:
Would `Span<char>` work for these? Then it could just use the strings on the config as-is rather than needing to call `ToArray()`?

Reply:
I think that the Span<char> only can be field inside a struct, not in a class because it cannot go to the heap.